.github/workflows/check.yaml

```diff
@@ -12,13 +12,13 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
-      - uses: subosito/flutter-action@v1
+      - uses: subosito/flutter-action@v2
+        with:
+          channel: 'stable'
+          cache: true
```

Comment:
Why is this here?

1. The [documentation for this action](https://github.com/subosito/flutter-action#usage) states that this interacts with [GitHub's `actions/cache`](https://github.com/actions/cache), which you're not using in this workflow, so it's doing nothing
2. GitHub runners are ephemeral so this would only work for this single workflow runtime instance anyway, so I fail to see what you're optimising for

Reply:
1. Ive checked the [example usage of this flow in the official repository](https://github.com/subosito/flutter-action/blob/main/.github/workflows/workflow.yml) and the `test_cache` job doesn't define `actions/cache` anywhere, so I thought that this is configured by the step itself, it also looks like that from the yaml perspective, since the code for runner looks like that:

```yaml
runs:
  using: 'composite'
  steps:
    - if: ${{ inputs.cache == 'true' }}
      uses: actions/cache@v2
      with:
        path: ${{ inputs.cache-path }}
        key: ${{ inputs.cache-key }}-${{ inputs.channel }}-${{ inputs.flutter-version }}
```

My thinking is, this should setup caching by itself, but it's possible my thinking isn't correct

Reply:
2. Currently, on each pull request we open 5 checks, and each of them redownloads the Flutter engine. My idea was to reduce the amount of downloads, by simply enabling the cache for flutter step. I'm not too familiar with GitHub actions, but as far as I understand, according to [official docs](https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows#about-caching-workflow-dependencies) these dependencies can be cached between build for 7 days. If that's not the case, then your point is valid and enabling the cache doesn't make sense

Reply:
1. Hmm. Curious. If it's 'composite', then why do we need to feed it in as an input? That doesn't make logical sense. 😕 
2. Oh, ok, it sounds like my understanding was incorrect regarding the ephemeral nature of the cache. It _does_ outlive that single workflow run. There's something to learn every day!

Reply:
I think the `composite` is just here so the action can use multiple steps - one for cache setup and one to run the script.